sdks/go/pkg/beam/core/runtime/exec/coder.go

```diff
@@ -1208,3 +1213,91 @@ func DecodeWindowedValueHeader(dec WindowDecoder, r io.Reader) ([]typex.Window,
 
 	return ws, t, pn, nil
 }
+
+// EncodeTimer encodes a typex.TimerMap into a byte stream.
+func EncodeTimer(elm ElementEncoder, tm typex.TimerMap, w io.Writer) error {
```

Comment:
Does this need to be exported?

Reply:
It doesn't need to.  I had both of these encode and decode in `graph/coder/` but it creates weird import cycle. To separate some parts, I think Encoding will work fine but we need to pass in exec.ElementDecoder and exec.WindowDecoder to Decode which creates weird import cycle.